features/docs/writing_support_code/hooks/before_all_hook.feature

```diff
@@ -0,0 +1,49 @@
+Feature: BeforeAll Hooks
+
+  BeforeAll hooks can be used if you have some set-up to be done before all
+  scenarios are executed.
+
+  BeforeAll hooks are not aware of your configuration. Use AfterConfiguration if
+  you need it.
+
+  Scenario: A single BeforeAll hook
+
+    A BeforeAll hook will be invoked a single time before all the scenarios are
+    executed.
+
+    Given a file named "features/f.feature" with:
+      """
+      Feature: BeforeAll hook
+        Scenario: #1
```

Comment:
Question. Is `#` in the scenario name permitted / going to cause problems.

Reply:
I'll check that 😅

Reply:
I review the scenario, and made some tests.
the `#` character seems to be properly supported in scenario names